boa_parser/src/parser/cursor/mod.rs

```diff
@@ -169,6 +179,22 @@ where
         self.private_environment_nested_index != 0
     }
 
+    /// Set the identifier of the cursor.
+    #[inline]
+    pub(super) fn set_identifier(&mut self, identifier: u32) {
+        self.identifier = identifier;
+    }
+
+    /// Get the identifier for a tagged template.
+    #[inline]
+    pub(super) fn tagged_template_identifier(&mut self) -> u64 {
+        let mut hasher = FxHasher::default();
+        hasher.write_u32(self.identifier);
+        hasher.write_u32(self.tagged_templates_count);
+        self.tagged_templates_count += 1;
+        hasher.finish()
+    }
```

Comment:
```suggestion
    pub(super) fn tagged_template_identifier(&mut self) -> u64 {
        let identifier = self.identifier as u64;
        let count = self.tagged_templates_count as u64;

        (count << 32) | identifier
    }
```

I meant something like this, where we have the the first `32bits` for the count and the last for the generation/identifier.

Reply:
Yeah true, just pushing the u64 is easier. Applied.